sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/AnsiCastSuiteBase.scala

```diff
@@ -25,6 +25,14 @@ import org.apache.spark.sql.internal.SQLConf
 import org.apache.spark.sql.types._
 import org.apache.spark.unsafe.types.UTF8String
 
+/**
+ * Test suite base for
+ *   1. [[Cast]] with ANSI mode enabled
+ *   2. [[AnsiCast]]
+ *   3. [[TryCast]]
```

Comment:
Maybe it's too late now, but why `TryCast` is tested with `AnsiCastSuiteBase`? it's not ansi actually.

Reply:
Yes `TryCast` is identical to CAST with configuration `spark.sql.ansi.enabled` as true. For the test cases in AnsiCastSuiteBase are suitable for `TryCastSuite` except that all the exceptions should not happen in the test suite. There is a hack for overriding `checkExceptionInExpression` in `TryCastSuite` to ensure that.